internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -428,43 +453,52 @@ func (r *BMACReconciler) reconcileAgentInventory(bmh *bmh_v1alpha1.BareMetalHost
 
 // Utility to verify whether a BMH should be reconciled based on the InfraEnv
 //
-// This function verifies three things:
+// This function verifies the following things:
 //
-// 1. InfraEnv existsD
+// 1. InfraEnv exists
 // 2. InfraEnv's ISODownloadURL exists
 // 3. The BMH.Spec.URL is the same to the InfraEnv's ISODownload URL
+// 4. The InfraEnv's ISO is at least 60 seconds old
 //
-// If the three checks above are true, then the reconcile won't happen
+// If all the checks above are true, then the reconcile won't happen
 // and the reconcileBMH function will return.
 //
 // Note that this function is also used to decide whether a BMH reconcile should be queued
 // or not. This will help queuing fewer reconciles when we know the BMH is not ready.
 //
-// The function returns 2 booleans:
+// The function returns 2 booleans, time.Duration and a string:
 //
 // 1. Should we proceed with the BMH's reconcile
 // 2. Should the full `Reconcile` stop as well
-func shouldReconcileBMH(bmh *bmh_v1alpha1.BareMetalHost, infraEnv *aiv1beta1.InfraEnv) (bool, bool) {
+// 3. If requeueing is needed, for how long should we back off
+// 4. If reconciling should not continue, a reason that will be printed in the log
+func shouldReconcileBMH(bmh *bmh_v1alpha1.BareMetalHost, infraEnv *aiv1beta1.InfraEnv) (bool, bool, time.Duration, string) {
```

Comment:
Starting to think this function should return a `reconcileResult` ... but let's keep that as an improvement for 4.9

Reply:
I agree, it's very ugly that in the caller of this function I need to adjust the type based on the value of one of the returned fields (sic!)